File: app/src/main/java/xyz/godi/budgetmanager/Utils/Converters.java
package xyz.godi.budgetmanager.Utils;

import android.arch.persistence.room.TypeConverter;

import java.util.Date;

/**
 *  Converters class to converts a custom class to and from a known type that Room can persist.
 */
public class Converters {
    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

}

File: app/src/main/java/xyz/godi/budgetmanager/Utils/Converters.kt
package xyz.godi.budgetmanager.Utils

import androidx.room.TypeConverter
import java.util.*

/**
 * Converters class to converts a custom class to and from a known type that Room can persist.
 */
class Converters {

    @TypeConverter fun calenderToDateStamp(calendar: Calendar): Long = calendar.timeInMillis

    @TypeConverter fun dateStampToCalendar(value: Long): Calendar =
            Calendar.getInstance().apply { timeInMillis = value }

}